libguard/guard_interface.cpp

```diff
@@ -247,55 +248,22 @@ GuardRecords getAll()
     return guardRecords;
 }
 
-/**
- * @brief Helper function to delete guard record
- *
- * @param[in] record to delete
- * @param[in] recordPos position for deleting record
- * @param[in] posOfLastRecord position of last record in guard file
- *
- * @return NULL on success
- *         Throw exception on failure
- *
- */
-static void deleteRecord(GuardRecord record, int recordPos, int posOfLastRecord)
-{
-    GuardRecord nullGuard;
-    size_t sizeOfGuardRecord = sizeof(nullGuard);
-
-    memset(&nullGuard, 0xFF, sizeOfGuardRecord);
-
-    uint32_t offset = recordPos * sizeOfGuardRecord;
-
-    GuardFile file(guardFilePath);
-    file.erase(offset + headerSize, sizeOfGuardRecord);
-
-    int i = recordPos + 1;
-    while (posOfLastRecord != i)
-    {
-        uint32_t offset1 = i * sizeOfGuardRecord;
-        file.read(offset1 + headerSize, &record, sizeOfGuardRecord);
-        uint32_t offset2 = (i - 1) * sizeOfGuardRecord;
-        file.write(offset2 + headerSize, &record, sizeOfGuardRecord);
-        file.write(offset1 + headerSize, &nullGuard, sizeOfGuardRecord);
-        i++;
-    }
-}
-
 void clear(const EntityPath& entityPath)
```

Comment:
Will take this later.

Reply:
This is valid command , just a copy of function, and log message also same here.

Reply:
Will take this in separate PR.

Reply:
since this is new function, should handle in this PR only.

Reply:
This function is from last PR. I didn't introduced this function in this PR. 
I told I will take care of making a common code later instead of handling in this PR.

Reply:
Done.